.github/workflows/lint.yml

```diff
@@ -0,0 +1,43 @@
+name: Lint Workflow
+on:
+  push:
+  schedule:
+    - cron: "0 0 * * 0"
+jobs:
+  clippy:
+    name: Clippy
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: clippy
+          profile: minimal
+          override: true
+      - run: cargo build
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: -- -D clippy::all -D warnings
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --features python-extension -- -D clippy::all -D warnings
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --features wasm-extension -- -D clippy::all -D warnings
+  rustfmt:
+    name: Format
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          components: rustfmt
+          profile: minimal
+          override: true
+      - run: cargo build
+      - run: cargo fmt --all -- --check
```

Comment:
why not add these jobs to the main workflow like in https://github.com/ripe-tech/ripe-core/blob/master/.github/workflows/main.yml

this way we avoid the building step if something is in the wrong format

Reply:
true, we prolly should have them split and have a depends_on on the yml, but for consistency, ill change this